config/crd/bases/ols.openshift.io_configs.yaml

```diff
@@ -0,0 +1,201 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.12.0
+  name: configs.ols.openshift.io
+spec:
+  group: ols.openshift.io
+  names:
+    kind: Config
+    listKind: ConfigList
+    plural: configs
+    singular: config
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: Config is the Schema for the configs API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: ConfigSpec defines the desired state of OLS deployment.
+            properties:
+              llm:
+                description: LLMSpecSpec defines the desired state of the large language
+                  model (LLM).
+                properties:
+                  providers:
+                    items:
+                      description: ModelSpec defines the desired state of LLM provider.
+                      properties:
+                        credentialPath:
+                          description: Provider credential path in filesystem
+                          type: string
+                        credentials:
+                          description: Provider credentials (via environment variables)
+                          type: string
```

Comment:
we don't want this either, credentials should be an object reference to a well-formed secret